tsdb/db_test.go

```diff
@@ -155,6 +155,65 @@ func TestDataAvailableOnlyAfterCommit(t *testing.T) {
 	testutil.Equals(t, map[string][]tsdbutil.Sample{`{foo="bar"}`: {sample{t: 0, v: 0}}}, seriesSet)
 }
 
+// TestNoPanicAfterWALCorrutpion ensures that querying the db after a WAL corruption doesn't cause a panic.
+// https://github.com/prometheus/prometheus/issues/7548
+func TestNoPanicAfterWALCorrutpion(t *testing.T) {
+	db, closeFn := openTestDB(t, &Options{WALSegmentSize: 32 * 1024}, nil)
+	defer func() {
+		closeFn()
+	}()
+
+	// Append untill we have at least one mmaped chunk.
+	var expSamples []tsdbutil.Sample
+	var maxt int64
+	{
+		app := db.Appender()
+		for {
+			_, err := app.Add(labels.FromStrings("foo", "bar"), maxt, 0)
+			testutil.Ok(t, err)
+			err = app.Commit()
+			testutil.Ok(t, err)
+			expSamples = append(expSamples, sample{t: maxt, v: 0})
+			files, err := ioutil.ReadDir(mmappedChunksDir(db.Dir()))
+			testutil.Ok(t, err)
+			if len(files) > 0 {
+				break
+			}
+			maxt++
+		}
+		testutil.Ok(t, db.Close())
+	}
+
+	// Corrupt the second wal to trigger a repair.
```

Comment:
How are we sure that there is a 2nd WAL file?

Reply:
Because otherwise the `walFiles[1].Name()` will panic.